store/store_helper.go

```diff
@@ -0,0 +1,57 @@
+package store
+
+import (
+	types "github.com/cosmos/cosmos-sdk/store/types"
+)
+
+// GetAndDecode gets and decodes key and returns it. Returns nil if key doesn't exist.
+func GetAndDecode[T any](store types.KVStore, dec func([]byte) (T, error), key []byte) (T, error) {
+	var res T
+	bz := store.Get(key)
+	if bz == nil {
+		return res, nil
+	}
+	resp, err := dec(bz)
+	if err != nil {
+		return resp, err
+	}
+	return resp, err
```

Comment:
```suggestion
	var res T
	bz := store.Get(key)
	if bz == nil {
		return res, nil
	}

	resp, err := dec(bz)
	if err != nil {
		return resp, err
	}

	return resp, nil
```

Reply:
Also, why do we have `var res T`? Can't we just return `nil` in that one case?

Reply:
no, we cannot use `nil` as `T` value type in return statement